In [1]:
s = "this is My First Python programming class and i am learNING python string and its function"

## Question-1 - Try to extract data from index one to index 300 with a jump of 3 

In [3]:
s[1:300:3]

'h   r tnrrmglsnimeNGyosi dtfco'

#Answer - 'h   r tnrrmglsnimeNGyosi dtfco'

## Question-2 - Try to reverse a string without using reverse function 

In [6]:
s[::-1]

'noitcnuf sti dna gnirts nohtyp GNINrael ma i dna ssalc gnimmargorp nohtyP tsriF yM si siht'

##Answer - 'noitcnuf sti dna gnirts nohtyp GNINrael ma i dna ssalc gnimmargorp nohtyP tsriF yM si siht'

## Question-3 - Try to split a string after conversion of entire string in uppercase 

In [7]:
s1 = s.upper()
s1.split()

['THIS',
 'IS',
 'MY',
 'FIRST',
 'PYTHON',
 'PROGRAMMING',
 'CLASS',
 'AND',
 'I',
 'AM',
 'LEARNING',
 'PYTHON',
 'STRING',
 'AND',
 'ITS',
 'FUNCTION']

##Answer - ['THIS',
 'IS',
 'MY',
 'FIRST',
 'PYTHON',
 'PROGRAMMING',
 'CLASS',
 'AND',
 'I',
 'AM',
 'LEARNING',
 'PYTHON',
 'STRING',
 'AND',
 'ITS',
 'FUNCTION']

## Question- 4 -  try to convert the whole string into lower case

In [8]:
s2 = s.lower()
s2

'this is my first python programming class and i am learning python string and its function'

##Answer - 'this is my first python programming class and i am learning python string and its function'

## Question-5 - Try to capitalize the whole string

In [11]:
s3 = s.capitalize()
s3

'This is my first python programming class and i am learning python string and its function'

##Answer - 'This is my first python programming class and i am learning python string and its function'

## Question-6 Write a diference between isalnum() and isalpha()

## ANSWER

##isalpha() method returns true if all characters of a string are only alphabets (between ‘A’ to ‘Z’ and between ‘a’ to ‘z’). If there is any non alphabet character in the string, method returns false.

In [17]:
s.isalpha()

False

In [18]:
"shefali".isalpha()

True

##isalnum() method returns true if all characters of a string are either alphabets (between ‘A’ to ‘Z’ and between ‘a’ to ‘z’) or numbers (0 to 9) or both, otherwise method returns false

In [19]:
s.isalnum()

False

In [20]:
"sahu".isalnum()

True

In [21]:
"1234".isalnum()

True

In [23]:
"shefali sahu".isalnum() #space contain thats why it is coming

False

## Question-7 - Try to give an example of expand tab

In [24]:
s3 = 'xyz\t12345\tabc'
result = s3.expandtabs()
result

'xyz     12345   abc'

##Answer - 'xyz     12345   abc'

## Question-8 - Give an example of strip , lstrip and rstrip 

##Answer - 

In [25]:
string = " shefali " #extra space at both the side
string.strip()

'shefali'

In [26]:
string = "shefali " #extra space at right side
string.rstrip()

'shefali'

In [27]:
string = " shefali" #extra space at left side
string.lstrip()

'shefali'

## Question-9 - Replace a string charecter by another charector by taking your own example 

##Answer-

In [1]:
string = 'pythonhxamples'
position = 6
new_character = 'e'
temp = list(string)
temp[position] = new_character
string = "".join(temp)

In [6]:
string

'pythonexamples'

## Question - 10 - Try  to give a defination of string center function with and exmple

##Answer -

center() method-is an inbuilt function that is used to align a string to the center by filling paddings to the left and right of the string.

In [4]:
s.center(500)

'                                                                                                                                                                                                             this is My First Python programming class and i am learNING python string and its function                                                                                                                                                                                                             '

## Question - 11 - Write your own definition of compiler and interpretor without copy paste form internet in your own language

##Answer

##Interpreter - An interpreter is a kind of program that executes other programs. When you write Python programs , it converts source code written by the developer into intermediate language which is again translated into the native language / machine language that is executed

##Compiler - Python compile() function takes source code as input and returns a code object which is ready to be executed and which can later be executed by the exec() function. 

## Question - 12 - Python is a interpreted of compiled language give a clear ans with your understanding 

##Answer

it's interpreted.
Python is an interpreted, object-oriented, high-level programming language...
Since there is no compilation step ...
The Python interpreter and the extensive standard library are available...
Instead, when the interpreter discovers an error, it raises an exception. When the program doesn't catch the exception, the interpreter prints a stack trace.

## Question -13 - Try to write a usecase of python with your understanding 

##Web application development,
Data Science,
Artificial Intelligence,
Game development,
Internet of Things,
Desktop GUI

## DONE BY SHEFALI SAHU ....THANKS